rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1133,7 +1133,7 @@ stream {
 
                 return 497;
                 {{ else }}
-                return {{ $all.Cfg.HTTPRedirectCode }} https://$best_http_host$request_uri;
+                return {{ $all.Cfg.HTTPRedirectCode }} https://$host$request_uri;
```

Comment:
I think we should stick to `$best_http_host` and trip `:<port>` instead if it exists.

Otherwise requests coming through an another reverse proxy with X-Forwarded-Host will get an incorrect `Location`.

Reply:
> I think we should stick to `$best_http_host` and trip `:<port>` instead if it exists.

Ok but that should be done in lua (runtime value). That's fine to you?

Reply:
Maybe something like

```
--- a/rootfs/etc/nginx/template/nginx.tmpl
+++ b/rootfs/etc/nginx/template/nginx.tmpl
@@ -1021,6 +1021,25 @@ stream {
             {{ end }}

             rewrite_by_lua_block {
+              {{/* redirect to HTTPS can be achieved forcing the redirect or having a SSL Certificate configured for the server */}}
+              {{ if (or $location.Rewrite.ForceSSLRedirect (and (not (empty $server.SSLCert.PemFileName)) $location.Rewrite.SSLRedirect)) }}
+              {{ if not (isLocationInLocationList $location $all.Cfg.NoTLSRedirectLocations) }}
+              # enforce ssl on server side
+              if ngx.var.redirect_to_https == 1 {
+                  {{ if $location.UsePortInRedirects }}
+                  -- using custom ports require a different rewrite directive
+                  {{ local redirect_port = (printf ":%v" $all.ListenPorts.HTTPS) }}
+                  -- I'm not sure about following
+                  -- error_page 497 ={{ $all.Cfg.HTTPRedirectCode }} https://$host{{ $redirect_port }}$request_uri;
+
+                  ngx.exit(497)
+                  {{ else }}
+                  return ngx.redirect("full url without port", {{ $all.Cfg.HTTPRedirectCode }})
+                  {{ end }}
+              }
+              {{ end }}
+              {{ end }}
+
                 balancer.rewrite()
             }

@@ -1121,24 +1140,6 @@ stream {
             set $proxy_upstream_name    "{{ buildUpstreamName $location }}";
             set $proxy_host             $proxy_upstream_name;

-            {{/* redirect to HTTPS can be achieved forcing the redirect or having a SSL Certificate configured for the server */}}
-            {{ if (or $location.Rewrite.ForceSSLRedirect (and (not (empty $server.SSLCert.PemFileName)) $location.Rewrite.SSLRedirect)) }}
-            {{ if not (isLocationInLocationList $location $all.Cfg.NoTLSRedirectLocations) }}
-            # enforce ssl on server side
-            if ($redirect_to_https) {
-                {{ if $location.UsePortInRedirects }}
-                # using custom ports require a different rewrite directive
-                {{ $redirect_port := (printf ":%v" $all.ListenPorts.HTTPS) }}
-                error_page 497 ={{ $all.Cfg.HTTPRedirectCode }} https://$host{{ $redirect_port }}$request_uri;
-
-                return 497;
-                {{ else }}
-                return {{ $all.Cfg.HTTPRedirectCode }} https://$best_http_host$request_uri;
-                {{ end }}
-            }
-            {{ end }}
-            {{ end }}
-
```